### PR TITLE
Add ipfs page info bubble

### PR DESCRIPTION
**Pull request by spylogsster**
Resolves https://github.com/brave/brave-browser/issues/14889

image

## Submitter Checklist:

- [x] https://github.com/brave/security/issues/384
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

- Open IPFS link
- Click by badge in address bar

### CODE REVIEW

**Review by bbondy — APPROVED** [review]

File: chromium_src/components/page_info/page_info_ui.cc [hunk 1]
```diff
@@ -3,12 +3,37 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this file,
  * You can obtain one at http://mozilla.org/MPL/2.0/. */
 
+#include "brave/components/ipfs/buildflags/buildflags.h"
 #include "brave/components/vector_icons/vector_icons.h"
+#include "components/page_info/page_info_ui.h"
+#include "components/grit/brave_components_strings.h"
```
Comment (mkarolin) [inline chromium_src/components/page_info/page_info_ui.cc:9]: Nit: this one can go behind the `BUILDFLAG(IPFS_ENABLED)` flag too

File: chromium_src/components/page_info/page_info_ui.h [hunk 2]
```diff
@@ -0,0 +1,23 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_CHROMIUM_SRC_COMPONENTS_PAGE_INFO_PAGE_INFO_UI_H_
+#define BRAVE_CHROMIUM_SRC_COMPONENTS_PAGE_INFO_PAGE_INFO_UI_H_
+#include "brave/components/ipfs/buildflags/buildflags.h"
```
Comment (mkarolin) [inline chromium_src/components/page_info/page_info_ui.h:8]: Nit: empty line above.

File: components/page_info/BUILD.gn [hunk 3]
```diff
@@ -4,5 +4,8 @@
 # You can obtain one at http://mozilla.org/MPL/2.0/.
 
 group("page_info") {
-  deps = [ "//brave/components/vector_icons" ]
+  deps = [
+    "//brave/components/vector_icons",
+    "//brave/components/ipfs"
```
Comment (mkarolin) [inline components/page_info/BUILD.gn:9]: Sorry, forgot to mention: also, I think, the suggested practice is to add deps for all includes directly, which also means adding a dep on `//brave/components/ipfs/buildflags`. So, maybe add
`import("//brave/components/ipfs/buildflags/buildflags.gni")`
and `//brave/components/ipfs/buildflags` to `deps`, WDYT?

**Comment by mkarolin** [comment]
The wording seems a bit repetitive with `Learn more` at the end. Maybe something like:
`InterPlanetary File System (IPFS) is a public network for peer-to-peer data storage and delivery. You can configure your IPFS settings at any time in the browser Settings. <link>Learn more</link> about Brave's IPFS integration.` cc: @bbondy

File: components/page_info/BUILD.gn (hunk 3 above)
Comment (mkarolin) [reply components/page_info/BUILD.gn:9]: Correction: importing `//brave/components/ipfs/buildflags/buildflags.gni` isn't needed.

Comment (mkarolin) [reply components/page_info/BUILD.gn:9]: Or rather, maybe:
<pre>
import("//brave/components/ipfs/buildflags/buildflags.gni")

group("page_info") {
  deps = [
  "//brave/components/ipfs/buildflags",
   "//brave/components/vector_icons",
 ]

  if (ipfs_enabled) {
    deps += [ "//brave/components/ipfs" ]
  }
}
</pre>

**Review by mkarolin — APPROVED** [review]
`chromium_src` changes LGTM

**Pull request merged**